compiler-java/src/com/redhat/ceylon/ceylondoc/CeylonDocTool.java

```diff
@@ -892,8 +892,7 @@ private void copySourceFiles() throws FileNotFoundException, IOException {
                 markup.around("script src='" + getResourceUrl(pkg, "ceylondoc.js") + "' type='text/javascript'"); 
                 markup.close("head");
                 markup.open("body", "pre data-language='ceylon' style='font-family: Inconsolata, Monaco, Courier, monospace'");
-                // XXX source char encoding
-                BufferedReader input = new BufferedReader(new InputStreamReader(pu.getUnitFile().getInputStream()));
+                BufferedReader input = new BufferedReader(new InputStreamReader(pu.getUnitFile().getInputStream(), getEncoding()));
```

Comment:
The code seems correct based on other uses of `getEncoding()` in the file. https://github.com/ceylon/ceylon/blob/e01f23c7e7da6c36d77debbebbd21af1443191bc/compiler-java/src/com/redhat/ceylon/ceylondoc/CeylonDocTool.java#L466 uses it to configure the typechecker.